drizzle/migrations/0003_clerk_auth_sync.sql

```
@@ -0,0 +1,398 @@
+--
+-- Security improvements applied:
+-- - PL/pgSQL parameter names prefixed with p_ to avoid column ambiguity
+-- - Explicit table-qualified column references (public.users.xxx)
+-- - SECURITY DEFINER with SET search_path = public, auth, pg_temp on all functions
+-- - Input validation on save_chat_with_messages
+-- - Input validation on sync_clerk_user (email format check)
+-- =============================================
+
+-- 1. Ensure clerk_user_id column exists and has an index
+ALTER TABLE public.users ADD COLUMN IF NOT EXISTS clerk_user_id TEXT UNIQUE;
+CREATE INDEX IF NOT EXISTS idx_users_clerk_user_id ON public.users(clerk_user_id);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)